src/Sentry.Unity.Native/SentryNativeBridge.cs

```diff
@@ -168,27 +180,93 @@ private static void nativeLog(int cLevel, string message, IntPtr args, IntPtr us
                 return;
             }
 
-            // If the message contains any "formatting" modifiers (that should be substituted by `args`), we need
-            // to apply the formatting. However, we cannot access C var-arg (va_list) in c# thus we pass it back to
-            // vsnprintf (to find out the length of the resulting buffer) & vsprintf (to actually format the message).
-            if (message.Contains("%"))
+            string? message = null;
+            try
+            {
+                // We cannot access C var-arg (va_list) in c# thus we pass it back to vsnprintf to do the formatting.
+                // For Linux, we must make a copy of the VaList to be able to pass it back...
+                if (_isLinux)
+                {
+                    var argsStruct = Marshal.PtrToStructure<VaListLinux64>(args);
+                    var formattedLength = 0;
+                    WithMarshalledStruct(argsStruct, argsPtr =>
+                    {
+                        formattedLength = 1 + vsnprintf_linux(IntPtr.Zero, UIntPtr.Zero, format, argsPtr);
+                    });
+
+                    WithAllocatedPtr(formattedLength, buffer =>
+                    WithMarshalledStruct(argsStruct, argsPtr =>
+                    {
+                        vsnprintf_linux(buffer, (UIntPtr)formattedLength, format, argsPtr);
+                        message = Marshal.PtrToStringAnsi(buffer);
+                    }));
+                }
+                else
+                {
+                    var formattedLength = 1 + vsnprintf_windows(IntPtr.Zero, UIntPtr.Zero, format, args);
+                    WithAllocatedPtr(formattedLength, buffer =>
+                    {
+                        vsnprintf_windows(buffer, (UIntPtr)formattedLength, format, args);
+                        message = Marshal.PtrToStringAnsi(buffer);
+                    });
+                }
+            }
+            catch (Exception err)
+            {
+                logger.Log(level, $"Exception in native log forwarder: {err}");
```

Comment:
Exception can be null, but prefer the formatting overload so there's no allocations:

```suggestion
                logger.LogError("Exception in native log forwarder.", err);
```

Reply:
committed locally